libs/ledger/include/ledger/chaincode/contract.hpp

```diff
@@ -138,9 +140,9 @@ class Contract : public Chargeable
   bool ParseAsJson(chain::Transaction const &tx, variant::Variant &output);
 
   template <typename T>
-  bool GetStateRecord(T &record, ConstByteArray const &key);
+  bool GetStateRecord(T &record, chain::Address const &address);
```

Comment:
I was not expecting this change why is the key always an address?

Reply:
We agreed that we'll leave it as is for now and introduce a generic overload as a follow up